core/src/main/java/org/infinispan/remoting/inboundhandler/GlobalInboundInvocationHandler.java

```diff
@@ -153,7 +154,7 @@ private void runReplicableCommand(ReplicableCommand command, Reply reply, boolea
       try {
          invokeReplicableCommand(command, reply, preserveOrder);
       } catch (Throwable throwable) {
-         if (throwable.getCause() != null && throwable instanceof CompletionException) {
+         if (throwable.getCause() != null && (throwable instanceof CompletionException || throwable instanceof CacheConfigurationException)) {
```

Comment:
I have checked all places with `wireDependencies` and it turns out only this one catches `IllegalLifecycleStateException` :D So I'm just wondering... maybe we could simplify this and only change the `wireDependencies` method (leaving this place untouched)?